language/move-vm/runtime/src/interpreter.rs

```diff
@@ -864,6 +1032,188 @@ impl Frame {
             })
     }
 
+    fn get_local_ty(&self, idx: u8) -> PartialVMResult<&ParanoidType> {
+        self.local_tys
+            .get(idx as usize)
+            .ok_or_else(|| PartialVMError::new(StatusCode::VERIFIER_INVARIANT_VIOLATION))
+    }
+
+    fn get_mut_local_ty(&self, idx: u8) -> PartialVMResult<&mut ParanoidType> {
+        self.local_tys
+            .get_mut(idx as usize)
+            .ok_or_else(|| PartialVMError::new(StatusCode::VERIFIER_INVARIANT_VIOLATION))
+    }
+
+    fn type_stack_transition(
```

Comment:
Yeah, this wasn't quite what I was expecting, though this may be easier? I'm not sure. 

I was imagining replacing all interactions with the stacks with macros 

`pop` and `push` and `popn`/`pushn`/`lastn`. 

This way you could do something like `let (ty, value) = pop!(interpreter)` And even swap out the type for `ty` with a `cfg` feature flag if we want to disable the paranoid mode